### ISSUE TITLE
Templates already in the dom

### ISSUE THREAD

**Issue by mrjjwright** [Freak613/stage0#14]
If I have a template with `#names` that I load statically into the dom and then want to memoize with stage0 can I just modify core stage0 to have a little function that takes the content from the template or should I do something else?   Also are there any performance pitfalls to be aware of in this scenario?

```
  function t(content) {
    content._refPaths = genPath(content);
    content.collect = walker;
    return content;
  }
```

**Comment by Freak613** [comment]
I've just added `compile` function exported from the library, that you can use for this case.
However, from how compiler is built right now, I think that you need to remove template node from the DOM, or it will continue to parse the whole document. 
But it's only my assumption, maybe TreeWalker will stop once ContentNode end is reached, so you need to test it first without removing.

https://github.com/Freak613/stage0/blob/master/index.js#L65

**Comment by Freak613** [comment]
From the performance perspective, except node removing, I see no pitfalls here.

**Comment by mrjjwright** [comment]
Wow, thanks so much!!  trying it out!

**Comment by mrjjwright** [comment]
Oh and also it's working just fine as far as I can tell although I will keep an eye on it.

**Comment by Freak613** [comment]
It's because stage0 doesn't support DocumentFragment's for now, which is the type of `template.content` dom node.
Fragments needs to be handled differently, because its contents disappears once node is added to the dom.
https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment

**Comment by Freak613** [comment]
And yes, you need to do the same for Template nodes.

**Comment by mrjjwright** [comment]
Makes sense, thanks for the quick and excellent help today on your excellent lib.

**Comment by mrjjwright** [comment]
Also for others make sure that you don't add spaces in your template as the first element, making the `firstChild` a textNode like I did. :)

**Comment by chang-zhao** [comment]
Good health to you, Pavel, and to all kind people anywhere!
I hope you wouldn't mind my newbie question.
I'm not a professional programmer, just want to make some wiki-like CMS (PHP+JS).
1. Is there some reading, Russian or English, in simple words, on how to use such techniques like this stage0?
2. For example, if I want elements on a page to be highly manipulated by user (moved, resized, collapsed etc) does it make sense to add them with JS rather than send them as HTML ready-made by PHP? Or is it much much less performant?
3. Because it seems it could make sense to make lexical translations browser-side, like sending "[[some/link]]" from server to browser, and turning it to actual link ("<a...>some/link</a>") by JS?
(Definitely when user (editing his text) types "[[link]]", it could make sense to transform the syntax on the fly with JS (substituting or complementing wiki syntax with actual HTML link); so may be it makes sense to build whole document with JS rather than send in HTML?)
Thank you!
